### DISCUSSION TITLE
Why my API route.ts is statically rendered ?

### DISCUSSION THREAD

**Discussion by AlexandreBardiaux** [vercel/next.js#62725] [Help] [answered]
### Summary

I have an API route `/api/practitioners/search`. And that route is pre-rendered as static content during the build. So when I call my route, I get the data as of the time of the build, not the latest data. I read and tried all the things mentionned in the caching documentation of NextJS 14 but nothing worked.

I found a workaround by adding `request.url;`. So my route.ts file was 
```
import { restClient } from "@/restClient";
import { NextResponse } from "next/server";

export async function GET() {
  const data = getDataFromBackend()

  return NextResponse.json(data);
}
```

and became
```
import { restClient } from "@/restClient";
import { NextResponse } from "next/server";

export async function GET(request: Request) {
  request.url;
  const data = getDataFromBackend()
 
  return NextResponse.json(data);
}
```

This way, it works fine. My route is dynamically rendered. The Fetch requests are cached, but I can simply add { cache: "no-store" } if I don't want it to be cached.

The fact that I "use" the request object (with `request.url`) makes sure that my route is dynamically rendered. Is there another (better) way to make this route dynamically and not statically rendered ? And why NextJS 14 statically renders a route ? If I call a route, it means that I want to trigger some action or get some data. In my opinion, a route should always be dynamically rendered. It is really confusing otherwise.

**Comment by icyJoseph** [comment, marked as answer]
Hi


You should be able to just do `export const dynamic = 'force-dynamic'` anywhere in the file.

- https://nextjs.org/docs/app/building-your-application/routing/route-handlers#opting-out-of-caching
- https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#dynamic

The first link talks about ways to opt-out from the caching, which is the default behavior. The last option listed is called route segment config, and there you'll find the second link, which talks about controlling this via the `dynamic` export.

You could also activate the Next.js TypeScript plugin, https://nextjs.org/docs/app/building-your-application/configuring/typescript#typescript-plugin, and that'll give you auto-complete when working with `export const dynamic`.

**Reply by AlexandreBardiaux** [reply]
Thanks for the clarification 👍

**Reply by ShakChunni** [reply]
Thank you!

**Reply by MahidunNobi** [reply]
@icyJoseph Thank you very  much for saving my day.
